Bill splitter v0.1

ejilonok opened the pull request:
Добавлена первая версия делителя счета
ЯП, Спринт 2

ejilonok committed:
Bill splitter v0.1
Добавлена первая версия делителя счета
ЯП, Спринт 2

faritowich commented on src/main/java/UserTextInterface.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("\tДобро пожаловать в программу разделения счета!");
+        System.out.println("Данная программа предназначена для ввода числа гостей и списка заказанных товаров или услуг.");
+        System.out.println("Когда вы введете полный перечень товаров или услуг, программа автоматически предоставит Вам список и сумму оплаты каждым гостем.");
+        System.out.println("-----------------------------------------------");
```
Такие подряд идущие println можно объединить в один.

ejilonok replied:
Спасибо. Обычно использую такую форму для улучшения читаемости кода.
В новом ПР поправила.

faritowich commented on src/main/java/UserTextInterface.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("-----------------------------------------------");
+    }
+
+    int getFriendCount() {
```
private

В других методах тоже стоит проверить

ejilonok replied:
По всем методам проставила модификаторы доступа.

faritowich commented on src/main/java/UserTextInterface.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("Спасибо, что воспользовались программой! Надеюсь, Вы довольны ее работой! До новых встреч ;)");
+        System.out.println("Bill splitter ver 0.1\t\tcreated by ejil");
+    }
+}
```
Разбивка логики хорошая, но имхо лучше разбить этот класс на несколько классов. Чтобы один отвечал за ввод людей, другой за ввод товаров, третий за добавление окончания итд

ejilonok replied:
Добавила классы в соответствии с рекомендацией

ejilonok committed:
Bill splitter v0.2
Добавлена вторая версия делителя счета.
Изменена логика работы программы. Добавлена разбивка по логическим шагам. Разбивка и пользовательского интерфейса, и логики работы программы.
Скорректирован ввод числа друзей. Вставлена обработка исключений
Скорректирован вывод рублей. Исправлена ошибка расчета.

ЯП, Спринт 2

ejilonok committed:
Bill splitter v0.2
Добавлена вторая версия делителя счета.
Изменена логика работы программы. Добавлена разбивка по логическим шагам. Разбивка и пользовательского интерфейса, и логики работы программы.
Скорректирован ввод числа друзей. Вставлена обработка исключений
Скорректирован вывод рублей. Исправлена ошибка расчета.

ЯП, Спринт 2
